x-pack/apm-server/aggregation/spanmetrics/aggregator.go

```diff
@@ -19,9 +21,38 @@ import (
 
 // AggregatorConfig holds configuration for creating an Aggregator.
 type AggregatorConfig struct {
-	Report   publish.Reporter
-	Logger   *logp.Logger
+	// Report is a publish.Reporter for reporting metrics documents.
+	Report publish.Reporter
+
+	// MaxGroups is the maximum number of distinct service destination
+	// group metrics to store within an aggregation period. Once this
+	// number of groups is reached, any new aggregation keys will cause
+	// individual metrics documents to be immediately published.
+	MaxGroups int
+
+	// Interval is the interval between publishing of aggregated metrics.
+	// There may be additional metrics reported at arbitrary times if the
+	// aggregation groups fill up.
 	Interval time.Duration
+
+	// Logger is the logger for logging metrics aggregation/publishing.
+	//
+	// If Logger is nil, a new logger will be constructed.
+	Logger *logp.Logger
+}
+
+// Validate validates the aggregator config.
+func (config AggregatorConfig) Validate() error {
+	if config.Report == nil {
+		return errors.New("Report unspecified")
+	}
+	if config.MaxGroups <= 0 {
+		return errors.New("MaxGroups unspecified or negative")
```

Comment:
btw, here I forgot to mention that this line is redundant and not reachable because of the `validate` tag. Same for interval and the ones in tx metrics